test/views/declaration/DucrEntryViewSpec.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.declaration
+
+import base.{ExportsTestData, Injector}
+import controllers.declaration.routes
+import controllers.declaration.routes.DeclarantDetailsController
+import forms.Ducr
+import forms.Ducr.form
+import models.DeclarationType._
+import models.requests.JourneyRequest
+import org.jsoup.nodes.Document
+import play.api.data.Form
+import views.declaration.spec.PageWithButtonsSpec
+import views.html.declaration.ducr_entry
+import views.tags.ViewTest
+
+@ViewTest
+class DucrEntryViewSpec extends PageWithButtonsSpec with Injector {
+
+  override val typeAndViewInstance = (STANDARD, page(Ducr.form)(_, _))
+
+  val incorrectDUCR = "7GB000000000000-1234512345123451234512345"
+  val page = instanceOf[ducr_entry]
+
+  private def createView(form: Form[Ducr] = Ducr.form)(implicit request: JourneyRequest[_]): Document =
+    page(form)(request, messages)
+
+  "Ducr Entry" should {
+
+    "have correct message keys" in {
+      messages must haveTranslationFor("declaration.ducrEntry.header")
```

Comment:
This is only my view on this but in general... if you already have a proper test where you indirectly are also testing the message, like here at line 61, what's the need for this? :hugs:

Reply:
I agree. Its a pattern that has been used across view tests well before I joined so its another instance of it here
Maybe to discuss in dev chat whether they need to be there?